### DISCUSSION TITLE
How to use an external executable (Sidecar) in Tauri and correctly load a dynamic link library (DLL)?

### DISCUSSION THREAD

**Discussion by Burial0268** [tauri-apps/tauri#11836] [Q&A]
Hi everyone,

I’m currently developing a Tauri app and plan to embed an external executable (sidecar) on the Windows platform. This executable depends on a dynamic link library (DLL), but I’m encountering difficulties ensuring that the DLL is loaded correctly.

I would like to ask for help regarding how to make sure the external executable can properly call its dependent DLL under the following circumstances:

1. **DLL Location**:
   - Should the DLL file be placed in the same directory as the executable?
   - Or should I configure the `PATH` environment variable to specify the DLL’s location?

2. **Configuration in Tauri**:
   - How do I correctly specify the path to the sidecar executable in the Tauri configuration file?
   - Are there recommended methods to ensure that the DLL can be found when Tauri starts the executable?

I’d appreciate any advice or shared experience that could help me resolve this issue.

Thanks in advance!

**Comment by FabianLars** [comment]
On Windows you can just bundle the files as resources instead, and spawn the sidecar as a normal Command.
> Should the DLL file be placed in the same directory as the executable?

yeah, that's the default way to do it on windows.

**Reply by raphaelmenges** [reply]
How to make Tauri also sign that DLL on bundling?

Related: <https://github.com/tauri-apps/tauri/discussions/12001>
